### ISSUE TITLE
Outputs "up to date" when dependency has new version available

### ISSUE THREAD

**Issue by daboross** [kbknapp/cargo-outdated#6]
`cargo outdated` shows that all dependencies are up to date when they are clearly not.

Here's an example:

```
$ cat Cargo.toml 
[package]
name = "outdated_test"
version = "0.1.0"
authors = ["Dabo Ross <daboross@daboross.net>"]

[dependencies]
pistoncore-window = "0.8"
$ cargo build
    Updating registry `https://github.com/rust-lang/crates.io-index`
 Downloading pistoncore-window v0.8.0
 Downloading pistoncore-input v0.4.0
   Compiling rustc-serialize v0.3.16
   Compiling bitflags v0.3.2
   Compiling piston-float v0.1.0
   Compiling shader_version v0.2.1
   Compiling piston-viewport v0.1.0
   Compiling pistoncore-input v0.4.0
   Compiling pistoncore-window v0.8.0
   Compiling outdated_test v0.1.0 (file:///home/daboross/outdated_test)
$ cargo outdated
Checking for SemVer compatible updates...Done
Checking for the latest updates...Done
All dependencies are up to date, yay!
$ cargo search "pistoncore-window"
    Updating registry `https://github.com/rust-lang/crates.io-index`
pistoncore-window (0.9.1)    A library for window abstraction
```

As you can see, a new version of `pistoncore-window` (0.9.1) is clearly available, but `cargo outdated` doesn't show this.

**Comment by kbknapp** [comment]
Yep I noticed this today too, I'll start investigating :) thanks for taking the time to file this!

**Comment by blaenk** [comment]
Ah, was excited to use this but yeah I noticed this too. Subscribing to this issue.

**Comment by kbknapp** [comment]
Yeah, unfortunately I've had a ton less time to work on this issue than I thought I would. I'm hoping to take a deep look this weekend as I'm assuming it's actually an easy fix and only due to some updates in `cargo update` which this subcommand uses on the backend.

I'll update here as I find things.

**Comment by Esption** [comment]
With https://github.com/rust-lang/rfcs/pull/1241 we'll no longer be able to have wildcards on crates.io, so it'll be nice to have this working.

The best I can work out in like 20 minutes of figuring out the codebase is that the Root Deps / All Deps stuff with `debug` on is getting the latest version rather than the one that Cargo.lock says.  Even in this, `clap` should be out-of-date (Cargo.lock says v1.2.1) but it says it's using the latest (v1.4.5).  I'm not exactly sure how any of this works as I've never touched semver and there's a painful lack of comments, but that's the best I can come up with.

**Comment by AerialX** [comment]
This was fixed by #7

**Comment by daboross** [comment]
I know it's a ton later, but I was just checking back on this and thank you for fixing this! Excellent command now, I'll be using it frequently!
